src/Elastic.CommonSchema/EcsDocument.cs

```diff
@@ -85,11 +90,12 @@ public partial class EcsDocument
 			Error = GetError(exception),
 			Service = GetService(initialCache)
 		};
-		SetActivityData(doc);
 
 		if (options?.IncludeHost is null or true) doc.Host = GetHost(initialCache);
 		if (options?.IncludeProcess is null or true) doc.Process = GetProcess(initialCache);
 		if (options?.IncludeUser is null or true) doc.User = GetUser();
+		if (options?.IncludeTraceId is null or true)
+			SetActivityData(doc);
```

Comment:
@Mpdreamz Should I make the suggested change in this pull-request? It will complicate a little since [ActivityExtensions.cs](https://github.com/elastic/ecs-dotnet/pull/src/Elastic.CommonSchema.NLog/ActivityExtensions.cs), since it will require shared code between projects (new public API)

Reply:
Let's do it in a follow up PR :+1:

It would be nice if they used the same logic across logging implementations. 

ToString already calls ToHexString however we currently have different fallbacks if the format is not W3C or if it's null or empty.